### PR TITLE
aleth-interpreter library

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: libaleth-interpreter/VM.h [hunk 1]
```diff
@@ -17,9 +17,10 @@
 
 #pragma once
 
-#include "Instruction.h"
 #include "VMConfig.h"
-#include "VMFace.h"
+
+#include <libevm/VMFace.h>
```
Comment (gumb0) [inline libaleth-interpreter/VM.h]: So libaleth-interpreter depends on libevm? This is going to change later, right?

File: libaleth-interpreter/VMConfig.h [hunk 2]
```diff
@@ -0,0 +1,433 @@
+/*
```
Comment (gumb0) [inline libaleth-interpreter/VMConfig.h]: Hm, why does github doesn't show this file as moved

Comment (gumb0) [reply libaleth-interpreter/VMConfig.h]: Did you change it?

Comment (gumb0) [reply libaleth-interpreter/VMConfig.h]: Ah is it copied?

File: libaleth-interpreter/VM.h (hunk 1 above)
Comment (chfast) [reply libaleth-interpreter/VM.h]: Only on headers. This should change, but is not a priority. Home helper functions are defined in VMFace.h. I didn't want to do a lot of changes when moving files.

At least this PR allow building aleth-interpreter as a shared library and distribute it in this form.

File: libaleth-interpreter/VMConfig.h (hunk 2 above)
Comment (chfast) [reply libaleth-interpreter/VMConfig.h]: It's a copy. The config was shared between legacy and interpreter VMs.

**Comment by chfast** [comment]
I believe there are 2 easy changes that can be done here later. Use instruction metrics from evmc library and do not include headers from libevm. Maybe third would be to remove some compile time optimization options (only leave the default values).

But we have to decide what to do with https://github.com/ethereum/cpp-ethereum/pull/5065 as these 2 are in conflict.

File: libaleth-interpreter/VMConfig.h (hunk 2 above)
Comment (axic) [reply libaleth-interpreter/VMConfig.h]: Perhaps you want to move the file to `libaleth-interpreter` and place a copy into the original directory?

That way if `libaleth-interpreter` is split out of the main tree using "git magic" the history will stay proper.

**Comment by axic** [comment]
The above is my only commit and I like the change FWIW. Would vote to merge this first, the other PR should detect if files are moved during a rebase and apply the changes properly.

File: libaleth-interpreter/VMConfig.h (hunk 2 above)
Comment (chfast) [reply libaleth-interpreter/VMConfig.h]: I don't think VMConfig is super important, but I can try to do it.

File: libevm/LegacyVMConfig.h [hunk 3]
```diff
@@ -0,0 +1,444 @@
+/*
```
Comment (chfast) [inline libevm/LegacyVMConfig.h]: This is a copy now.

**Pull request merged**